### DISCUSSION TITLE
Context deadline exceeded error and unable to clone

### DISCUSSION THREAD

**Discussion by LPnave** [fluxcd/flux2#2196] [Q&A] [answered]
Hi,

Flux is unable to reconcile the git repo that I have created and it kept giving me  _"context deadline exceeded"_ error and I was even unable to delete that source from flux. So I restarted the source controller container(pod) and I was able to delete the source. 

But now flux is unable to clone the git source. It would say _"unable to clone: remote authentication required but no callback set"_
I am using **Azure DevOps** and I have used a **PAT** as mentioned in the documentation

Please help me fix this

Thank you

**Comment by kingdonb** [comment, marked as answer]
Please provide more information about the version of Flux that you have in use.

There are some very particular bits about how Azure DevOps works; the last I checked you must use `libgit2` provider to interface with Azure DevOps. Is your gitrepo source configured to use that implementation?

Here is a doc: https://fluxcd.io/docs/components/source/gitrepositories/#git-implementation

If you deleted the source, it seems likely this `gitImplementation` configuration detail might have been lost. This is something you probably shouldn't need to know, and we're discussing how Flux should be able to detect the protocol on your behalf and save us from needing to provide that, so it is opaque to the user.

But currently, that is not the case. Please check the doc above and let us know if it helps!
